sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -984,10 +986,28 @@ def custom_metric(x1, x2):
 
     dist1, ind1 = nbrs1.kneighbors(X)
     dist2, ind2 = nbrs2.kneighbors(X)
-
+    
+    assert_true(nbrs1._fit_method, 'ball_tree')
+    assert_true(nbrs2._fit_method, 'ball_tree')
+    
     assert_array_almost_equal(dist1, dist2)
 
 
+def test_algo_auto_metrics():
+    X = rng.rand(12, 3)
+    Xcsr = csr_matrix(X)
+  
+    for metric in VALID_METRICS['brute']:
+        nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
+                                        metric=metric).fit(X)
+        assert_true(nn._fit_method, 'brute')
```

Comment:
shouldn't it be enough to make sure no error is raised? I don't understand this test. The other metrics might also support this metric.